Spanner/tests/Unit/OperationTest.php

```diff
@@ -287,6 +289,37 @@ public function testReadWithSnapshot()
         $this->assertEquals(self::TRANSACTION, $res->snapshot()->id());
     }
 
+    public function testReadWithCustomValueMapper()
+    {
+        $mapperProphesy = $this->prophesize(ValueMapper::class);
+        $mapperProphesy->decodeValues(
+            Argument::any(),
+            Argument::any(),
+            Argument::any()
+        )->shouldBeCalled()->willReturn(['ID' => '99']);
+
+        $mapperDummy = $mapperProphesy->reveal();
+
+        $this->connection->streamingRead(Argument::allOf(
+            Argument::withEntry('table', 'Posts'),
+            Argument::withEntry('session', self::SESSION),
+            Argument::withEntry('keySet', ['all' => true]),
+            Argument::withEntry('columns', ['foo'])
+        ))->shouldBeCalled()->willReturn($this->executeAndReadResponse());
+
+        $operation = TestHelpers::stub(Operation::class, [
+            $this->connection->reveal(),
+            false,
+            null,
+            $mapperDummy,
+        ]);
+
+        $res = $operation->read($this->session, 'Posts', new KeySet(['all' => true]), ['foo']);
+        $this->assertInstanceOf(Result::class, $res);
+        $rows = iterator_to_array($res->rows());
+        $this->assertSame('99', $rows[0]['ID']);
```

Comment:
This is failing in the tests right now (only on ubuntu, for some reason) with `Failed asserting that 10 is identical to '99'.`